config/traefik.yaml

```diff
@@ -0,0 +1,113 @@
+entrypoints:
+  # Defines a secure entrypoint using TLS encryption
+  websecure:
+    address: ":443"
+    http:
+      tls: true
+  # Defines an insecure entrypoint that redirects to the secure one.
+  web:
+    address: ":80"
+    http:
+      # Redirects HTTP to HTTPS
+      redirections:
+        entrypoint:
+          to: "websecure"
+          scheme: "https"
+
+api:
+  insecure: true
+  dashboard: true
+
+http:
+  services:
+    grist:
+      loadBalancer:
+        servers:
+          - url: 'http://grist:8484'
+    dex:
+      loadBalancer:
+        servers:
+          - url: 'http://dex:5556'
+    authelia:
+      loadBalancer:
+        servers:
+          - url: 'http://authelia:9091'
+
+  middlewares:
+    no-fwd:
+      headers:
+        customRequestHeaders:
+          'X-Forwarded-User': ""
+
+  routers:
+    route-grist-login:
+      rule: "PathPrefix(`/auth/login`) || PathPrefix(`/_oauth`)"
+      service: grist
+      entryPoints:
+        - web
+
+    route-grist:
+      rule: "PathPrefix(`/`)"
+      priority: 1   # Set a lower priority than the other rules
+      service: grist
+      middlewares:
+        - no-fwd
+      entryPoints:
+        - web
+
+    route-dex:
+      rule: "PathPrefix(`/dex/`) || Path(`/dex`)"
+      service: dex
+      entryPoints:
+        - web
+
+    route-authelia:
+      rule: "PathPrefix(`/authelia/`) || Path(`/authelia`)"
+      service: authelia
+      entryPoints:
+        - web
+
+{{ $use_https := env "USE_HTTPS" }}
+{{if eq $use_https "true" }}
+    https-route-grist-login:
+      rule: 'Host(`{{ env "APP_HOST" }}`) && (PathPrefix(`/auth/login`) || PathPrefix(`/_oauth`))'
```

Comment:
This `env "APP_HOST"`, is the idea here that we pass through the domain with a `https://` or `http://` prefix? 

I guess I'm not quite sure what purpose this serves in this rule, and am requesting enlightenment!

Reply:
It's supposed to be `GRIST_HOST`, was a vestige from Omnibus. Since we're using HTTPS, the rule has to [match the hostname](https://doc.traefik.io/traefik/routing/routers/#host-and-hostregexp).

Fixed the variable name.